src/components/World/World.tsx

```diff
@@ -0,0 +1,57 @@
+import { useOutletContext } from 'react-router-dom';
+import { Tooltip } from 'react-tooltip';
+
+import { data } from './data';
+
+import { IUser } from '../../interfaces/user';
+import { getDifferenceInDays } from '../../utils/date';
+
+import './world.scss';
+
+export default function World() {
+  const [user] = useOutletContext() as [user: IUser];
+
+  const daysInEveryCountry = user.travelHistory.reduce((acc, country) => {
+    const days = getDifferenceInDays(country.from, country.to || new Date().toString());
+    acc[country.countryCode] = acc[country.countryCode] ? acc[country.countryCode] + days : days;
+    return acc;
+  }, {} as Record<string, number>);
+
+  const maxDaysInSingleCountry = Object.values(daysInEveryCountry).reduce((acc, value) => {
+    acc = Math.max(acc, value);
+    return acc;
+  }, 0);
+
+  function getOpacity(countryCode: string) {
+    if (!daysInEveryCountry[countryCode]) {
+      return false;
+    }
+
+    const minOpacity = 0.5;
+    const halfPossibleOpacity = daysInEveryCountry[countryCode] / maxDaysInSingleCountry / 2;
+    const finalOpacity = +(halfPossibleOpacity + minOpacity).toFixed(2);
```

Comment:
if we don't use half opacity we might utilize `+((possibleOpacity + minOpacity) % 1).toFixed(2)`
but you should test if it works properly